keras_cv/models/segmentation/deeplab_v3/deeplab_v3.py

```diff
@@ -0,0 +1,255 @@
+        (ECCV 2018)
+        - [Rethinking Atrous Convolution for Semantic Image Segmentation](https://arxiv.org/abs/1706.05587)
+        (CVPR 2017)
+
+    Args:
+        num_classes: int, the number of classes for the detection model. Note
+            that the `num_classes` doesn't contain the background class, and the
+            classes from the data should be represented by integers with range
+            [0, `num_classes`).
+        backbone: `keras.Model`. The backbone network for the model that is
+            used as a feature extractor for the DeepLabV3 Encoder. Should be a
+            `keras_cv.models.backbones.backbone.Backbone`. A somewhat sensible
+            backbone to use in many cases is the:
+            `keras_cv.models.ResNet50V2Backbone.from_preset("resnet50_v2_imagenet")`.
+        use_low_level_features: bool, whether to combine low-level features from
+            the backbone with the output of the encoder or not. If set to `True`,
+            the model is DeepLabV3+, otherwise it is DeepLabV3.
+        projection_filters: int, number of filters in the
+            convolution layer projecting low-level features from the `backbone`.
+            The default value is set to `48`, as per the
+            [TensorFlow implementation of DeepLab](https://github.com/tensorflow/models/blob/master/research/deeplab/model.py#L676).
+            This parameter is only relevant if `use_low_level_features` is also
+            specified.
+        spatial_pyramid_pooling: (Optional) a `keras.layers.Layer`. Also known
+            as Atrous Spatial Pyramid Pooling (ASPP). Performs spatial pooling
+            on different spatial levels in the pyramid, with dilation. If
+            provided, the feature map from the backbone is passed to it inside
+            the DeepLabV3 Encoder, otherwise
+            `keras_cv.layers.spatial_pyramid.SpatialPyramidPooling` is used.
+        segmentation_head: (Optional) a `keras.layers.Layer`. If provided, the
+            outputs of the DeepLabV3 encoder is passed to this layer and it
+            should predict the segmentation mask based on feature from backbone
+            and feature from decoder, otherwise a similar architecture is used.
+
+    Examples:
+    ```python
+    import tensorflow as tf
+    import keras_cv
+
+    images = tf.ones(shape=(1, 96, 96, 3))
+    labels = tf.zeros(shape=(1, 96, 96, 1))
+    backbone = keras_cv.models.ResNet50V2Backbone(input_shape=[96, 96, 3])
+    model = keras_cv.models.segmentation.DeepLabV3(
+        num_classes=1, backbone=backbone,
+    )
+
+    # Evaluate model
+    model(images)
+
+    # Train model
+    model.compile(
+        optimizer="adam",
+        loss=keras.losses.BinaryCrossentropy(from_logits=False),
+        metrics=["accuracy"],
+    )
+    model.fit(images, labels, epochs=3)
+    ```
+    """
+
+    def __init__(
+        self,
+        num_classes,
+        backbone,
+        use_low_level_features=False,
```

Comment:
Is there any reason users would want to exclude the low level features? It seems to me like everyone probably wants to use DeepLabV3Plus instead of DeepLabV3, but maybe there are use cases?